platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/NodeId.java

```diff
@@ -72,4 +128,37 @@ public int getIdAsInt() {
     public String toString() {
         return Long.toString(id);
     }
+
+    @Override
+    public void serialize(SerializableDataOutputStream out) throws IOException {
+        out.writeLong(id);
+    }
+
+    @Override
+    public void deserialize(SerializableDataInputStream in, int version) throws IOException {
+        id = in.readLong();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
```

Comment:
Any particular reason to call `this.getClass()`, rather than just hardcoding `NodeId.class`?

Reply:
it was auto generated code.   I'll change it.

Reply:
It's probably because NodeId isn't final, someone could extend it without overriding the equals.